scrapyd_k8s/log_handler.py

```diff
@@ -0,0 +1,161 @@
+import os
+import threading
+import tempfile
+import logging
+
+from kubernetes import client, watch
+from .object_storage import LibcloudObjectStorage
+
+logger = logging.getLogger(__name__)
+
+watcher_threads = {}
+pod_tmp_mapping = {}
+
+def get_last_n_lines(file_path, num_lines):
+    """Efficiently retrieve the last `num_lines` lines from `file_path`."""
+    try:
+        with open(file_path, 'rb') as f:
+            # Move to the end of the file
+            f.seek(0, os.SEEK_END)
```

Comment:
Ah, great you're using seek here!

Reply:
I had to spend some time to make this new algorithm work:D Reading chunks of bytes is a bit more sophisticated that reading lines in Python:)